pkg/prometheus/promcfg_test.go

```diff
@@ -8632,231 +8632,107 @@ scrape_configs:
 	require.Equal(t, expected, string(cfg))
 }
 
-func TestScrapeConfigStaticConfig(t *testing.T) {
-	p := defaultPrometheus()
-
-	cg := mustNewConfigGenerator(t, p)
-	cfg, err := cg.GenerateServerConfiguration(
-		p.Spec.EvaluationInterval,
-		p.Spec.QueryLogFile,
-		p.Spec.RuleSelector,
-		p.Spec.Exemplars,
-		p.Spec.TSDB,
-		p.Spec.Alerting,
-		p.Spec.RemoteRead,
-		nil,
-		nil,
-		nil,
-		map[string]*monitoringv1alpha1.ScrapeConfig{
-			"probe1": {
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "testscrapeconfig1",
-					Namespace: "default",
-					Labels: map[string]string{
-						"group": "group1",
-					},
-				},
-				Spec: monitoringv1alpha1.ScrapeConfigSpec{
-					StaticConfigs: []monitoringv1alpha1.StaticConfig{
-						{
-							Targets: []monitoringv1alpha1.Target{"http://localhost:9100"},
-							Labels: map[monitoringv1.LabelName]string{
-								"label1": "value1",
-							},
+func TestScrapeConfigSpecConfig(t *testing.T) {
+	refreshInterval := monitoringv1.Duration("5m")
+	for _, tc := range []struct {
+		name        string
+		patchProm   func(*monitoringv1.Prometheus) *monitoringv1.Prometheus
```

Comment:
I've added this field to the test struct in case some test cases need to patch the default Prometheus, for instance, `EnforcedNamespaceLabel`. I had this idea when refactoring the Probes tests

Reply:
I've modified it once more to be, no need for an assignment since we pass Prometheus as a pointer
```go 
patchProm   func(*monitoringv1.Prometheus)
```